frontend/ui/widget/touchmenu.lua

```diff
@@ -217,15 +225,44 @@ function TouchMenuBar:init()
                 }
             }
             for i, sep in ipairs(self.icon_seps) do
-                local current_icon = i == k - 1 or i == k
-                self.icon_seps[i].style = current_icon and "solid" or "none"
+                local current_icon = (i == k - 1 or i == k) and k ~= #self.icons
+                local last_icon = i == k and k == #self.icons
+                sep.style = current_icon and "solid" or "none"
+
+                -- if this is last icon then the empty bar segment has to move over to the right
+                -- by the width of a separator and the stretch width
+                if last_icon then
+                    self.bar_sep.empty_segments = {
+                        {
+                            s = icon_sep_width + stretch_width + _start_seg, e = icon_sep_width + stretch_width + _end_seg
+                        }
+                    }
+                    self.icon_seps[i].style = "solid"
```

Comment:
`self.icon_seps[i]` can be replaced by `sep` here? `sep.style = current_icon and "solid" or "none"` can also be moved into the else block after this if block.